api/service/syncing/syncing.go

```diff
@@ -898,18 +899,36 @@ func (ss *StateSync) GetMaxPeerHeight() uint64 {
 }
 
 // IsOutOfSync checks whether the node is out of sync from other peers
-func (ss *StateSync) IsOutOfSync(bc *core.BlockChain) bool {
+func (ss *StateSync) IsOutOfSync(bc *core.BlockChain, doubleCheck bool) bool {
```

Comment:
to avoid premature trigger of sync in a bad timing. During consensus in a short time window, some nodes will have the block added while other don't (due to network delay). If the node checks sync status during this window, it will enter sync mode and stop all consensus process, which is bad. This change made sure the out of sync status is really the case rather than false positive.